generate.go

```diff
@@ -0,0 +1,203 @@
+package cogs
+
+import (
+	"fmt"
+
+	"github.com/pelletier/go-toml"
+)
+
+type Cog struct {
+	Name string
+	// generate multiple envs?
```

Comment:
I guess we have to define what a "cog" is. It feels like its not set in stone yet, the way I thought about it was a "cog" is either each of the env (which sounds weird now that i type it down), or a "cog" is each of the moving parts for the env. 

If the latter, what we have doesnt make sense, but seeing how the command was used I think a cog is the env itself.  If the cog is the env, then I think this makes sense as we process one at a time.

Not sure if this made sense at all lol let me know, I can elaborate. I think what you have now makes sense, but it depends how we define a "cog"

Reply:
That's a fair point, I honestly slapped it on to see how it sounded. it makes more sense to have each manifest file be the cog, I don't have a name yet for an env slice of that manifest but "switching gears" sounds cool in terms of jumping environments.

Reply:
> "switching gears" sounds cool in terms of jumping environments.

Yeap!I like that actually, in that case we can leave it as is and see if it becomes unworkable, but feels alright for now

Reply:
[This is an interesting thread](https://ell.stackexchange.com/questions/235386/what-is-the-difference-between-cog-and-gear#answer-235423). So we can treat the term "gear" as an operating state rather than gear the object, that means a gear is an "operating state" of a cog

Reply:
That is very interesting indeed, didnt know the technical differences between the two. I can roll with this